buildSrc/src/main/kotlin/org/cqfn/save/buildutils/VersioningConfiguration.kt

```diff
@@ -2,20 +2,44 @@
  * Version configuration file.
  */
 
+@file:Suppress("UNUSED_IMPORT")
+
 package org.cqfn.save.buildutils
 
-import org.ajoberstar.reckon.gradle.ReckonExtension
-import org.ajoberstar.reckon.gradle.ReckonPlugin
+import com.palantir.gradle.gitversion.GitVersionPlugin
+import com.palantir.gradle.gitversion.VersionDetails
+import groovy.lang.Closure
+import org.gradle.api.GradleException
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.apply
-import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.extra
+import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.provideDelegate
 
-@Suppress("MISSING_KDOC_ON_FUNCTION", "MISSING_KDOC_TOP_LEVEL")
+/**
+ * Configures how project version is determined. We are using `git-version` plugin to get a version from git repo.
+ * If working tree is dirty (i.e. there are uncommitted or untracked changes) and release build is attempted, build will fail.
+ *
+ * @throws GradleException if there was an attempt to run release build with dirty working tree
+ */
 fun Project.configureVersioning() {
-    apply<ReckonPlugin>()
+    require(this == rootProject) { "Versioning should be configured for the root project" }
+
+    apply<GitVersionPlugin>()
+
+    val versionDetails: Closure<VersionDetails> by extra
+    val details = versionDetails.invoke()
+
+    allprojects {
+        version = details.version.let {
+            if (it.startsWith("v")) it.trim('v') else it
```

Comment:
why not to check that versioning is correct?
check for the semver format

Reply:
Added a pattern, requiring that tags start with `'v'`